uwblueprint/project-read-frontend
# Add sessions page & configure navbar routing

### Notion ticket link



[Sessions page shell](https://www.notion.so/uwblueprintexecs/9cb454262ed84c6299c0682c23f92ec5?v=1076bd30711a4a01838becee05f8b5a3&p=6d0068f63e9f4c5e9568e21b6e9d8795)
[Navbar shell](https://www.notion.so/uwblueprintexecs/9cb454262ed84c6299c0682c23f92ec5?v=1076bd30711a4a01838becee05f8b5a3&p=e7b0e9967f984489ae74dad29a566866)



### Implementation description

- created a navbar component that links to the registration & sessions pages
- refactored some routing stuff
  - fixed bug where going to /login directly would break because the Login component checks for a referer that doesn't exist
  - created a ProjectREAD component to house all logged-in pages, and provides the field context to them
- added a sessions page
  - removed session-specific code from the registration component
  - registered the private route in the ProjectREAD component



### Steps to test

1. yarn start
2. verify that:
    - if unauthed, going to / or /sessions redirects to /login, and redirects back to the original url after logging in
    - if authed, going to /login redirects to /
    - navbar links work as expected
    - going to a url that doesn't exist (ex. /avocados) redirects to /login if unauthed, / if authed
    - sessions page:

<img width="1535" alt="Screen Shot 2021-04-05 at 7 37 20 PM" src="https://user-images.githubusercontent.com/37346399/113639052-5867b200-9646-11eb-9007-00ccae952e54.png">



### What should reviewers focus on?

- testing functionality & code correctness

### Checklist
